plugins/observer/main.go

```diff
@@ -0,0 +1,325 @@
+	}
+	ds := client.Dataset(dataset)
+	err = ds.Create(ctx, nil)
+	if err != nil && !isAlreadyExists(err) {
+		return err
+	}
+
+	tbl := ds.Table(tableName)
+	err = tbl.Create(ctx, &bigquery.TableMetadata{Schema: schema})
+	if err != nil && !isAlreadyExists(err) {
+		return err
+	}
+
+	return nil
+}
+
+func isAlreadyExists(err error) bool {
+	var gerr *googleapi.Error
+	if errors.As(err, &gerr) {
+		if gerr.Code == http.StatusConflict {
+			return true
+		}
+	}
+	return false
+}
+
+// {
+// 	organization(login: "gitpod-io") {
+// 	  project(number: 3) {
+// 		columns(first: 100) {
+// 		  nodes {
+// 			name
+// 			cards(first: 100) {
+// 			  nodes {
+// 				content {
+// 				  ... on Issue {
+// 					id
+// 					number
+// 					title
+// 				  }
+// 				  ... on PullRequest {
+// 					id
+// 					number
+// 					title
+// 				  }
+// 				}
+// 			  }
+// 			}
+// 		  }
+// 		}
+// 	  }
+// 	}
+//   }
+
+type queryIssue struct {
+	Number    githubql.Int
+	Title     githubql.String
+	Assignees struct {
+		Nodes []struct {
+			Login githubql.String
+		}
+	} `graphql:"assignees(first: 5)"`
+}
+
+type projectQuery struct {
+	Organisation struct {
+		Project struct {
+			Name    githubql.String
+			Columns struct {
+				Nodes []struct {
+					Name  string
+					Cards struct {
+						Nodes []struct {
+							Content struct {
+								PR    queryIssue `graphql:"... on PullRequest"`
+								Issue queryIssue `graphql:"... on Issue"`
+							}
+						}
+					} `graphql:"cards(first: 100)"`
+				}
+			} `graphql:"columns(first: 6)"`
+		} `graphql:"project(number: $prj)"`
+	} `graphql:"organization(login: $org)"`
+}
+
+func observeRepo(ctx context.Context, wg *sync.WaitGroup, tbl *bigquery.Table, gh github.Client, org string, projectNumber int, prj ManagedProject) {
+	defer wg.Done()
+
+	ins := tbl.Inserter()
+	vars := map[string]interface{}{
+		"prj": githubql.Int(projectNumber),
+		"org": githubql.String(org),
+	}
+
+	tick := time.NewTicker(5 * time.Minute)
+	defer tick.Stop()
+	for ; true; <-tick.C {
+		if ctx.Err() != nil {
+			break
+		}
+
+		now := time.Now()
+
+		var q projectQuery
+		qctx, cancel := context.WithTimeout(ctx, 60*time.Second)
+		err := gh.Query(qctx, &q, vars)
+		cancel()
+		if err != nil {
+			logrus.WithError(err).WithFields(vars).Error("cannot query project")
+			continue
+		}
+
+		for _, col := range q.Organisation.Project.Columns.Nodes {
+			var excluded bool
+			for _, c := range prj.ExcludeColumns {
+				if c == col.Name {
+					excluded = true
+					break
+				}
+			}
+			if excluded {
+				continue
+			}
+
+			for _, crd := range col.Cards.Nodes {
+				var issue queryIssue
+				if crd.Content.Issue.Title != "" {
+					issue = crd.Content.Issue
+				} else if crd.Content.PR.Title != "" {
+					issue = crd.Content.PR
+				} else {
+					continue
+				}
+
+				assignees := make([]string, len(issue.Assignees.Nodes))
+				for i := range issue.Assignees.Nodes {
+					assignees[i] = string(issue.Assignees.Nodes[i].Login)
```

Comment:
Would it make sense to distinguish between issue assignees and PR assignees (reviewers) for the analysis?